modules/inactive-users/class-inactive-users.php

```diff
@@ -41,7 +43,13 @@ public static function init() {
 		add_filter( 'determine_current_user', [ __CLASS__, 'record_activity' ], 30, 1 );
 
 		add_action( 'admin_init', [ __CLASS__, 'register_release_date' ] );
-		add_action( 'set_user_role', [ __CLASS__, 'user_promoted' ] );
+
+		// skipping inactivity checks for new users
+		if ( is_multisite() ) {
+			add_action( 'add_user_to_blog', [ __CLASS__, 'maybe_skip_inactivity_check_for_new_user' ] );
+		}
+		add_action( 'user_register', [ __CLASS__, 'maybe_skip_inactivity_check_for_new_user' ] );
+
```

Comment:
We're now skipping a user only when it's being added, and not when their roles are changed.